### PR TITLE
wlrctl: init at 0.2.1

### PR DESCRIPTION
**Pull request by Artturin**
#123848 but with myself added to maintainers and commits combined
Thanks to @puffnfresh for making the original pr

i tested these commands https://git.sr.ht/~brocellous/wlrctl#features-and-examples and they work

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125579` at 7bd5d711 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/b90ef7c8be41f607440298582cb9c098">wlrctl</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by Artturin** [comment]
I'll check if I can get it to compile on aarch64 later

File: pkgs/tools/wayland/wlrctl/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description = "Command line utility for miscellaneous wlroots Wayland extensions";
+    homepage = "https://git.sr.ht/~brocellous/wlrctl";
+    license = licenses.mit;
+    maintainers = with maintainers; [ puffnfresh artturin ];
+    platforms = platforms.linux;
+  };
+}
+
```
Comment (legendofmiracles) [inline pkgs/tools/wayland/wlrctl/default.nix:25]:
```suggestion
```

File: pkgs/tools/wayland/wlrctl/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+stdenv.mkDerivation rec {
+  pname = "wlrctl";
+  version = "0.2.1";
+
+  src = fetchFromSourcehut {
+    owner = "~brocellous";
+    repo = "wlrctl";
+    rev = "v${version}";
+    sha256 = "039cxc82k7x473n6d65jray90rj35qmfdmr390zy0c7ic7vn4b78";
+  };
```
Comment (legendofmiracles) [inline pkgs/tools/wayland/wlrctl/default.nix:12]:
```suggestion
  };
  
```

File: pkgs/tools/wayland/wlrctl/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  dontUseCmakeConfigure = true;
+  nativeBuildInputs = [ cmake meson pkg-config scdoc ninja ];
+  buildInputs = [ libxkbcommon wayland ];
+
+  meta = with lib; {
+    description = "Command line utility for miscellaneous wlroots Wayland extensions";
+    homepage = "https://git.sr.ht/~brocellous/wlrctl";
+    license = licenses.mit;
+    maintainers = with maintainers; [ puffnfresh artturin ];
+    platforms = platforms.linux;
```
Comment (legendofmiracles) [inline pkgs/tools/wayland/wlrctl/default.nix:22]: Wayland also runs on other unix operating systems

File: pkgs/tools/wayland/wlrctl/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.2.1";
+
+  src = fetchFromSourcehut {
+    owner = "~brocellous";
+    repo = "wlrctl";
+    rev = "v${version}";
+    sha256 = "039cxc82k7x473n6d65jray90rj35qmfdmr390zy0c7ic7vn4b78";
+  };
+  dontUseCmakeConfigure = true;
+  nativeBuildInputs = [ cmake meson pkg-config scdoc ninja ];
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wlrctl/default.nix:14]: Do we really need ninja, meson, cmake and pkg-config?

Comment (Artturin) [reply pkgs/tools/wayland/wlrctl/default.nix]: removed cmake

**Comment by SuperSandro2000** [comment]
@Artturin can you take a look at the build failure https://logs.nix.ci/?key=nixos/nixpkgs.125579&attempt_id=fa53ac21-ef55-4723-8496-d4f476febbc8 ?

**Comment by Artturin** [comment]
https://todo.sr.ht/~brocellous/wlrctl/4

**Comment by SuperSandro2000** [comment]
You could do something like 

https://github.com/NixOS/nixpkgs/blob/a74c1d6458ed9873ffba7aac80b6e500dc28e188/pkgs/development/libraries/mapbox-gl-native/default.nix#L48

```
NIX_CFLAGS_COMPILE = "-Wno-error=type-limits";
```

**Comment by Artturin** [comment]
got my raspi4 setup, will try to compile

**Comment by Artturin** [comment]
was just about to post 
 
```
 Result of `nixpkgs-review pr 125579` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wlrctl</li>
  </ul>
</details>
```

**Comment by deviant** [comment]
Uh, this should not be marked no-error. The warnings are there for a reason! This is now masking a logic bug on AArch64. The correct fix is probably to cast to unsigned char & check if the value is > 127.

**Pull request merged**